tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/dav/DavClient.java

```diff
@@ -150,15 +152,22 @@ private Mono<Void> handleContactCreationResponse(HttpClientResponse response, St
         };
     }
 
-    public Flux<VEvent> getCalendarVEventsByUid(String userId, String eventUid, String username) {
+    public Mono<Calendar> getVCalendarContainingVEvent(String userId, String eventUid, String username) {
         Preconditions.checkArgument(StringUtils.isNotEmpty(eventUid), "VEvent id should not be empty");
         Preconditions.checkArgument(StringUtils.isNotEmpty(userId), "OpenPaas user id should not be empty");
 
         return findUserCalendars(userId, username)
-            .flatMap(calendarURI -> getCalendarVEventsByUidFromCalendar(calendarURI, eventUid, username));
+            .flatMap(calendarURI ->
+                getVCalendarContainingVEventFromCalendar(calendarURI, eventUid, username)
+                    .switchIfEmpty(Mono.fromRunnable(
+                        () -> LOGGER.trace("VEvent '{}' was not found in Calendar '{}'.", eventUid, calendarURI)))
+                    .doOnError(error -> LOGGER.debug("Error while querying '{}' for VEvent '{}': ",
+                        calendarURI, eventUid, error))
+                    .onErrorResume(ex -> Mono.empty()))
```

Comment:
Do the log in onErrorResume this allows having a single reactor operator and is thus less costly without hurting readability.